### ISSUE TITLE
[BUG] Inconsistent results between training and eval metrics when training a session-based model 

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/models#961]
### Bug description

Currently, we can obverse that there is a big difference between metrics coming from `fit()` and `model.evaluate()` in this [example notebook](https://github.com/NVIDIA-Merlin/models/blob/main/examples/usecases/transformers-next-item-prediction.ipynb).  We can see that the model.evaluate() metric results are `4X` higher than those coming from training step. In the example we are using `SequenceMaskRandom` in the training phase, i.e. in the model.fit().

To investigate if `SequenceMaskLast` is working correctly, we also trained a model using `SequenceMaskLast` as pre arg in the `model.fit()` and then we checked out the `model.evaluate()` results using train set as well but the metric results did not match.

```
model.compile(run_eagerly=False, optimizer='adam', loss="categorical_crossentropy")
model.fit(train, batch_size=512, epochs=2, pre=mm.SequenceMaskLast(schema=seq_schema, target=target))

model.evaluate(
    train,
    batch_size=512,
    pre=mm.SequenceMaskLast(schema=seq_schema, target=target),
    return_dict=True
)
```
We assume the issue might be due to two issues:

- `SequenceMaskLast` does not work as expected
-  calculation of metrics are not correct either in `model.fit()` or in `model.evaluate()` step

### Steps/Code to reproduce bug

Please run this code [example notebook](https://github.com/NVIDIA-Merlin/models/blob/main/examples/usecases/transformers-next-item-prediction.ipynb) to repro the issue. you can replace  `SequenceMaskRandom` in the model.fit() with `SequenceMaskLast` as well.

### Expected behavior

- we want to know that all masking methods are working fine under the hood and we can get consistent results from model.fit() and model.evaluate(). 

### Environment details
merlin-tensorflow:22.12 image with the latest main branches are pulled.

**Comment by rnyak** [comment]
solved by https://github.com/NVIDIA-Merlin/models/pull/969
